From R.Marek@sh.cvut.cz Wed Sep 28 00:37:36 2005
Date: 27 Sep 2005 21:54:51 -0000
Message-ID: <20050927215451.19403.qmail@desitka.sh.cvut.cz>
From: <R.Marek@sh.cvut.cz>
To: greg@kroah.com
Subject: PCI: ICH6 ACPI and GPIO quirk
Cc: <michal.mleczko@gmail.com>

This patch just adds ACPI and GPIO regions to its LPC bridge, similar
way as ICH4 did. I would like to thank Michal Mleczko for testing.

Signed-off-by: Rudolf Marek <r.marek@sh.cvut.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 drivers/pci/quirks.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

--- gregkh-2.6.orig/drivers/pci/quirks.c	2005-09-28 15:26:39.000000000 -0700
+++ gregkh-2.6/drivers/pci/quirks.c	2005-09-28 15:26:48.000000000 -0700
@@ -339,6 +339,18 @@
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_INTEL,    PCI_DEVICE_ID_INTEL_82801EB_0,		quirk_ich4_lpc_acpi );
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_INTEL,    PCI_DEVICE_ID_INTEL_ESB_1,		quirk_ich4_lpc_acpi );
 
+static void __devinit quirk_ich6_lpc_acpi(struct pci_dev *dev)
+{
+	u32 region;
+
+	pci_read_config_dword(dev, 0x40, &region);
+	quirk_io_region(dev, region, 128, PCI_BRIDGE_RESOURCES);
+
+	pci_read_config_dword(dev, 0x48, &region);
+	quirk_io_region(dev, region, 64, PCI_BRIDGE_RESOURCES+1);
+}
+DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_INTEL,	PCI_DEVICE_ID_INTEL_ICH6_1, quirk_ich6_lpc_acpi );
+
 /*
  * VIA ACPI: One IO region pointed to by longword at
  *	0x48 or 0x20 (256 bytes of ACPI registers)
